### ISSUE TITLE
Localization with Areas

### ISSUE THREAD

**Issue by boppercr** [dotnet/AspNetCore.Docs#20564]
I've been searching a lot on how to use Localization and Resource Files when Areas are used. I have multiple cases in the web app we're creating where I need to have localizable content but it's located in an area. A concrete example we have is a controller in an area "Company". The path to access the controller is then : "Solution/Areas/Company/Controllers/HomeController.cs". My resources folder has the following structure and doesn't seem to work : "Resources/Areas/Company/Controllers/HomeController.resx" and it doesn't seem to find my resource file. Any help in regards to that? 

Also, maybe unrelated, but the _resourceBaseName of the ResourceManager that is created is "Solution..Areas.Company.etc.". Are the two dots normal behaviour?  

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d1833676-ff1b-c587-cc9c-81afffa8ef68
* Version Independent ID: ad989395-3f49-4d55-521e-cf6fe56281ee
* Content: [Globalization and localization in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/localization.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/localization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@hishamco is there a trick to areas?

**Comment by boppercr** [comment]
Hi guys! Thank you for the help and sorry for the disturbance, but I solved my issue.
My main problem all came from this bad boy : 
`services.AddLocalization(options =>
            {
                options.ResourcesPath = "/Resources";
            });`
Removing the "/" resolved my problem. 

For those curious, my Resource file structure is : "[Root]/Resources/Areas/{MyAreaName}/Controllers/{MyControllerName}.resx.

In summary, I "copied" the same folder structure of my areas. 

Again, thank you for your help.

Might I suggest updating the doc to include the areas as a possible scenario. I may not be the only one that face this question.

**Comment by Rick-Anderson** [comment]
@hishamco  should we update and say something about areas?

@boppercr  what do you think we should add to the doc for areas?

**Comment by boppercr** [comment]
In the section "Resource file naming", something along the lines of : "The relative path from your root to the file you want localized must be reflected in the Resources folder." And maybe follow that by a couple of examples that are most likely to happen, like a Model, a View with and without areas.
"/Resources/Models/Home/IndexViewModel.resx"
"/Resources/Views/Home/Home.resx"
"/Resources/Areas/MyArea/Controllers/HomeController.resx"
"/Resources/Areas/MyArea/Models/Home/IndexViewModel.resx"

Also, in the same section, it might be good to specify not putting a "/" in front of the `ResourcePath = "Resources"` since it will break everything.

Thanks again.

**Comment by Rick-Anderson** [comment]
@boppercr  thanks for the feedback and glad you solved your problem.

**Comment by hishamco** [comment]
Ya no need to have a `/`, so no action needed here @Rick-Anderson
